src/components/LanguageSelector/LanguageSelector.tsx

```diff
@@ -13,10 +16,31 @@ export const LanguageSelector = ({ language }: Props) => {
 	const { styles } = useStyles(styleSheet);
 	const [selectedLanguage, setSelectedLanguage] = useState<UserPreferredLanguage>(language);
```

Comment:
Where does the language value come from? If it comes from backend, you should not create this "inner" state. You should invalidate query, responsible for fetching this language value. If you want UI to "feel" faster, you can implement optimistic update. Please check this link `https://tanstack.com/query/latest/docs/framework/react/guides/optimistic-updates`. In case you will have any questions, please feel free to find me.

PS. IF language values comes from some kind of global state, eg. zustand, please update this global state in `onSuccess` mutation callback